### PR TITLE
docs: aa-core providers connection methods

### PR DESCRIPTION
**Pull request by denniswon**

### CODE REVIEW

File: site/packages/aa-core/provider/connect.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      content: connect
+  - - meta
+    - name: description
+      content: Overview of the connect method on ISmartAccountProvider
+  - - meta
+    - property: og:description
+      content: Overview of the connect method on ISmartAccountProvider
+---
+
+# `connect<TAccount extends ISmartContractAccount>`
```
Comment (avasisht23) [inline site/packages/aa-core/provider/connect.md:15]: do we want to call out the generic?

File: site/packages/aa-core/provider/connect.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Returns
+
+### `ISmartAccountProvider`
+
+The provider with the account connected
+
+## Parameters
+
+### `fn: (provider: PublicErc4337Client<TTransport> | PublicErc4337Client<HttpTransport>) => TAccount`
```
Comment (avasisht23) [inline site/packages/aa-core/provider/connect.md:39]: if writing generic, should say what TTransport extends

Comment (avasisht23) [reply site/packages/aa-core/provider/connect.md:39]: and prob TAccount here too?

File: site/packages/aa-core/provider/getAddress.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+<<< @/snippets/provider.ts
+
+:::
+
+## Returns
+
+### `Promise<Address>`
+
+A Promise that resolves to the address of the connected account.
```
Comment (avasisht23) [inline site/packages/aa-core/provider/getAddress.md:37]: nit: only b/c this is a dependent clause, not a sentence
```suggestion
A Promise that resolves to the address of the connected account
```

File: site/packages/aa-core/provider/isConnected.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+<<< @/snippets/provider.ts
+
+:::
+
+## Returns
+
+### `boolean`
+
+The boolean flag indicating if the account is connected.
```
Comment (avasisht23) [inline site/packages/aa-core/provider/isConnected.md:37]: nit: only b/c this is a dependent clause, not a sentence
```suggestion
The boolean flag indicating if the account is connected
```

**Review by avasisht23 — APPROVED** [review]
questions, but non-blocking lgtm

File: site/packages/aa-core/provider/connect.md (hunk 1 above)
Comment (avasisht23) [reply site/packages/aa-core/provider/connect.md:15]: @moldy530

Comment (denniswon) [reply site/packages/aa-core/provider/connect.md:15]: I'll remove the generics. think it just causes confusion.

Comment (moldy530) [reply site/packages/aa-core/provider/connect.md:15]: It depends. I've added them when I reference them in other parts of the doc. I think here it's fine to remove

Comment (denniswon) [reply site/packages/aa-core/provider/connect.md:15]: hmm but the return type of TAccount that extends ISmartContractAccount, so tricky to just remove. thoughts? @moldy530 @avasisht23

Comment (denniswon) [reply site/packages/aa-core/provider/connect.md:15]: nvm, changed to `SmartAccountProvider` and `BaseSmartContractAccount`

**Pull request merged**